### ISSUE TITLE
Scalar type definitions in PHP 8.0+ must use static methods and can't inherit from CustomScalarType

### ISSUE THREAD

**Issue by galileogal** [webonyx/graphql-php#904]
I've just installed PHP 8.0 and suddenly some of my custom scalar types stopped working. I've been using templates form: [https://github.com/stagemteam/php-graphql-scalars](url) and the were fine up until now.

It turned out that CustomScalarType calls methods statically, like this:

```php
public function serialize($value)
{
     return $this->config['serialize']($value);
}
```

where `$this->config['serialize']` contains respective class and method names. In my case this was for instance:

`[0] => 'graphqlDefs\type\scalar\DateType'`, `[1] => 'serialize'`

This worked fine until PHP 7+, but now - with PHP 8 - it throws an error:

`Error(code: 0): Non-static method graphqlDefs\\type\\scalar\\DateType::serialize() cannot be called statically`

When I tried to declare serialize method in my custom scalar type as static there was an error since it was declared as non-static in CustomScalarType from which my DateType was derived. Removing inheritance from CustomScalarType on my scalar type classes and declaring `serialize`, `parseValue` and `parseLiteral` as static methods solved the problem. 

The question is if it's the best way to do this. Or should CustomScalarType be redesigned?

**Comment by spawnia** [comment]
I do not think we can change the behaviour of `CustomScalarType` to accomodate instance methods in a backwards-compatible way. Thus, I documented its behaviour in https://github.com/webonyx/graphql-php/pull/905/files#diff-729cf2a4ccbbb374803478cd14b9dc46160e9432a11f1bd63e301389b45af135R102-R103.
